src/System.Management.Automation/DscSupport/JsonDeserializer.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Management.Automation.Runspaces;
+
+namespace Microsoft.PowerShell.DesiredStateConfiguration.Json
+{
+    internal class JsonDeserializer
+    {
+        #region Constructors
+
+        /// <summary>
+        /// Instantiates a default deserializer.
+        /// </summary>
+        public static JsonDeserializer Create()
+        {
+            return new JsonDeserializer();
+        }
+
+        #endregion Constructors
+
+        #region Methods
+
+        /// <summary>
+        /// Returns schema of Cim classes from specified json file.
+        /// </summary>
+        public IEnumerable<PSObject> DeserializeClasses(string json, bool useNewRunspace = false)
+        {
+            if (string.IsNullOrEmpty(json))
+            {
+                throw new ArgumentNullException(nameof(json));
+            }
+
+            IEnumerable<PSObject> result = null;
+            System.Management.Automation.PowerShell powerShell = null;
+
+            if (useNewRunspace)
+            {
+                // currently using RunspaceMode.NewRunspace will reset PSModulePath env var for the entire process
+                // this is something we want to avoid in DSC GuestConfigAgent scenario, so we use following workaround
+                var s_iss = InitialSessionState.CreateDefault();
+                s_iss.EnvironmentVariables.Add(
+                    new SessionStateVariableEntry(
+                        "PSModulePath",
+                        Environment.GetEnvironmentVariable("PSModulePath"),
+                        description: null));
+                powerShell = System.Management.Automation.PowerShell.Create(s_iss);
+            }
+            else
+            {
+                powerShell = System.Management.Automation.PowerShell.Create(RunspaceMode.CurrentRunspace);
+            }
+
+            using (powerShell)
+            {
+                const string convertFromJson = @"ConvertFrom-Json";
```

Comment:
Longer term if Json cmdlets are part of some other web module, I wonder if we should use System.Text.Json instead here?

Reply:
That is one option, yes.
Though long term we plan to have all dsc-related code (including this one) out of SMA :), in which case DSC module should not have problems having a dependency on a module where json cmdlets will end up.

In any case, switching from json cmdlet that is used here to using .NET json API directly is an isolated work item and can be done separately.

On a side note, it would be interesting to see perf comparision between our current json-cmdlets and new .NET json APIs...